libdrgn/type.c

```diff
@@ -197,6 +204,7 @@ void drgn_int_type_init(struct drgn_type *type, const char *name, uint64_t size,
 	}
 	type->_private.size = size;
 	type->_private.is_signed = is_signed;
+	type->_private.num_template_parameters = 0;
```

Comment:
Don't bother initializing this for types without template parameters. It should be caught by asserting in the getter.